pkg/client/singleflight.go

```diff
@@ -0,0 +1,178 @@
+package client
+
+import (
+	"context"
+	"encoding/base64"
+	"time"
+
+	"github.com/miekg/dns"
+	"golang.org/x/sync/singleflight"
+
+	"darvaza.org/resolver/pkg/errors"
+)
+
+var (
+	_ Client = (*SingleFlight)(nil)
+)
+
+const (
+	// DefaultSingleFlightExpiration tells how long will we cache
+	// the result after an exchange
+	DefaultSingleFlightExpiration = time.Second
```

Comment:
here I would prefer at least `DefaultSingleFlightExpiration = 1*time.Second`, if not a full `DefaultSingleFlightExpiration = time.Duration(1) * time.Second`

Reply:
amended as requested